mailet/standard/src/main/java/org/apache/james/transport/matchers/SenderHostIs.java

```diff
@@ -43,24 +46,23 @@
  * @version 1.0.0, 2002-09-10
  */
 public class SenderHostIs extends GenericMatcher {
-    /**
-     * The collection of host names to match against.
-     */
-    private Collection<String> senderHosts;
 
-    /**
-     * Initialize the mailet.
-     */
+    private Collection<String> senderHosts;
+    
     public void init()  {
-        //Parse the condition...
-        StringTokenizer st = new StringTokenizer(getCondition(), ", ", false);
+        String condition = getCondition();
+        Preconditions.checkNotNull(condition, "'condition' should not be null");
 
-        //..into a vector of domain names.
-        senderHosts = new java.util.HashSet<String>();
-        while (st.hasMoreTokens()) {
-            senderHosts.add(st.nextToken().toLowerCase(Locale.US));
-        }
-        senderHosts = Collections.unmodifiableCollection(senderHosts);
+        senderHosts =fillTheDomainsList(condition);
+    }
+
+    @VisibleForTesting Collection<String> fillTheDomainsList(String condition) {
+        //Parse the condition into a vector of domain names.
+        senderHosts = ImmutableSet.<String> builder()
```

Comment:
You are affecting senderHosts twice: once in this method, and once in the calling method.